sdk/spring/azure-spring-boot/src/test/java/com/azure/spring/aad/webapp/AzureOauth2RefreshTokenGrantRequestEntityConverterTest.java

```diff
@@ -0,0 +1,89 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.spring.aad.webapp;
+
+import org.junit.jupiter.api.Test;
+import org.springframework.boot.test.context.FilteredClassLoader;
+import org.springframework.boot.test.context.assertj.AssertableWebApplicationContext;
+import org.springframework.boot.test.context.runner.WebApplicationContextRunner;
+import org.springframework.http.HttpEntity;
+import org.springframework.http.RequestEntity;
+import org.springframework.security.oauth2.client.endpoint.OAuth2RefreshTokenGrantRequest;
+import org.springframework.security.oauth2.client.registration.ClientRegistration;
+import org.springframework.security.oauth2.core.OAuth2AccessToken;
+import org.springframework.security.oauth2.core.OAuth2RefreshToken;
+import org.springframework.security.oauth2.server.resource.BearerTokenAuthenticationToken;
+import org.springframework.util.MultiValueMap;
+
+import java.time.Instant;
+import java.time.temporal.ChronoUnit;
+import java.util.Optional;
+
+import static org.junit.jupiter.api.Assertions.assertEquals;
+import static org.junit.jupiter.api.Assertions.assertNull;
+
+public class AzureOauth2RefreshTokenGrantRequestEntityConverterTest {
+
+    private AADWebAppClientRegistrationRepository clientRepo;
+    private AzureOauth2RefreshTokenGrantRequestEntityConverter converter;
+    private ClientRegistration azure;
+    private ClientRegistration graph;
+
+    private final WebApplicationContextRunner contextRunner = new WebApplicationContextRunner()
+        .withClassLoader(new FilteredClassLoader(BearerTokenAuthenticationToken.class))
+        .withUserConfiguration(AADWebAppConfiguration.class)
+        .withPropertyValues("azure.activedirectory.authorization-server-uri = fake-uri",
+            "azure.activedirectory.authorization.graph.scopes = Calendars.Read",
+            "azure.activedirectory.client-id = fake-client-id",
+            "azure.activedirectory.client-secret = fake-client-secret",
+            "azure.activedirectory.tenant-id = fake-tenant-id",
+            "azure.activedirectory.user-group.allowed-groups = group1, group2");
```

Comment:
Can we reuse this code fragment instead of just copy it? For example: create an util class.